quint/test/types/constraintSolver.test.ts

```diff
@@ -192,6 +234,33 @@ describe('unify', () => {
     result.map(subs => assert.deepEqual(substitutionsToString(subs), '[ a |-> int, b |-> bool ]'))
   })
 
+  it('unifies sum-type', () => {
+    // We do not provide
+    const result = unify(
+      table,
+      sumType([
+        ['A', { kind: 'var', name: 'a' }],
+        ['B', { kind: 'int' }],
+        ['C', unitType(0n)],
+      ]),
+      sumType([
+        ['C', unitType(0n)],
+        ['A', { kind: 'str' }],
+        ['B', { kind: 'var', name: 'b' }],
+      ])
+    )
+
+    assert.isTrue(result.isRight())
+    result.map(subs => assert.deepEqual(substitutionsToString(subs), '[ a |-> str, b |-> int ]'))
+  })
+
+  // it('unifies sum type with open row', () => {
+  //   const result = unify(table, parseTypeOrThrow('A(a)'), parseTypeOrThrow('A(int) | B(str)'))
+
+  //   assert.isTrue(result.isRight())
+  //   result.map(subs => assert.deepEqual(substitutionsToString(subs), '[]'))
+  // })
```

Comment:
Should this be uncommented or removed?

Reply:
Removed! It was part of debugging. Thanks :)